RFC/src/RFC-0120_Consensus.md

```diff
@@ -0,0 +1,212 @@
+* The genesis block MUST have a height of 1.
+
+#### Prev_hash
+
+This is the hash of the previous block's header.
+
+The prev_hash MUST confirm to the following:
+
+* represented as an array of unsigned 8-bit integers (bytes).
+* MUST be a hash of the entire contents of the previous block's header.
+
+#### Timestamp
+
+This is the timestamp at which the block was mined.
+
+The timestamp MUST confirm to the following:
+
+* Must be transmitted as UNIX timestamp.
+* MUST be less than [FTL].
+* MUST be higher than the [MTP].
+
+#### Output_mr
+
+This is the merkle root of the outputs. This is calculated in the following way: Hash (txo MMR root  || roaring bitmap hash of UTXO indices).
+
+The output_mr MUST confirm to the following:
+
+* Represented as an array of unsigned 8-bit integers (bytes).
+* The hashing function used MUST be blake2b with a 256 bit digest.
+
+#### Range_proof_mr
+
+This is the merkle root of the range proofs.
+
+The range_proof_mr MUST confirm to the following:
+
+* Represented as an array of unsigned 8-bit integers (bytes).
+* The hashing function used must be blake2b with a 256 bit digest.
+
+#### Kernel_mr
+
+This is the merkle root of the outputs.
+
+The kernel_mr MUST confirm to the following:.
+
+* Must be transmitted as an array of unsigned 8-bit integers (bytes).
+* The hashing function used must be blake2b with a 256 bit digest.
+
+#### Total_kernel_offset
+
+This is total summed offset of all the transactions contained in this block.
+
+The total_kernel_offset MUST confirm to the following:
+
+* Must be transmitted as an array of unsigned 8-bit integers (bytes)
+
+#### Total_difficulty
+
+This is the total accumulated difficulty of the mined chained.
+
+The total_difficulty MUST confirm to the following:
+
+* Must be transmitted as unsigned 64-bit integer;
+* MUST be larger than the previous block's `total_difficulty`;
+
+#### Nonce
+
+This is the nonce used in solving the Proof of Work. 
+
+The nonce MUST confirm to the following:
+
+* Must be transmitted as unsigned 64-bit integer;
+
+#### PoW
+
+This is Proof of Work algorithm that was used to solve the Proof of Work. This is used in conjunction with the Nonce
+
+The [PoW] MUST confirm to the following:
+
+* Must be transmitted as <TO BE CONFIRMED>;
+
+
+### FTL
+
+The Future Time Limit. This is how far into the future a time is accepted as a valid time. Any time that is more than the FTL is rejected until such a time that it is not less than the FTL.
+The FTL is calculated as (T*N)/20 with T and N defined as:
+T: Target time - This is the ideal time that should pass between blocks which have been mined.
+N: Block window - This is the amount of blocks used when calculating difficulty adjustments.
+
+### MTP
+
+The Median Time Past. This is the lower limit of a time. Any time that is less than the MTP is rejected.
+THe MTP is calculated as the average timestamp of the previous 11 blocks.
```

Comment:
Median is surely better.
Average is not robust and could be gamed by malicious miners.